src/cpu/xbyak/xbyak_util.h

```diff
@@ -54,6 +54,7 @@
 */
 #include "xbyak.h"
 
+#ifndef NON_X86_CPU
```

Comment:
Please no changes in xbyak.h. They will be probably overwritten next time we update it.

Reply:
We have to change it as it has __inline asm inside. We could use ifndef (__x86_64__) to guard those inline asm.

Reply:
`cpu_isa_traits.hpp` is the unique place that `xbyak/` headers are included. So gen-dnn skips `xbyak/` entirely in [gen-dnn's cpu_isa_traits.hpp](https://github.com/necla-ml/gen-dnn/blob/master/src/cpu/cpu_isa_traits.hpp). My version has some unused crap in it that can be cleaned up for a PR). I did not need (or want) any inline asm in gen-dnn, because I compile it on x86 and two non-x86 processors and it passes gtests and benchdnn tests. I have not tried older x86 CPUs.  Would you even need any inline asm within `cpu_isa_traits.hpp`?

Most `src.cpu/` JIT code has some "reference" plain-C version .... _except_ .... Ahaa, I remember.

Last time I merged the messiest work to circumvent JIT code was in implementing `cpu_barrier.cpp`.  In this case, I wrote a barrier based on `__atomic_add_fetch` that seemed ok, even with one compiler that did not have the <atomic> header.  And even though `__atomic_add_fetch` seems pretty generic, you still may want to modify the spin loop.  (I use `__builtin_ia32_pause()` for some compiles, and `asm("nop":::)` for some non-x86 CPUs.  Your compiler/CPU may have its own preference for how to write an efficient spin loop. My `#ifdef`s would need to be cleaned up before merging into mkl-dnn, though.

@rsdubtso Would adding a ref impl for cpu_barrier fall into the "too much" category?  It's a fairly localized change.  Only a few lines of code if mkl-dnn could have _just_ the `__builtin_ia32_pause()` version.  And it would be nice if someone verified that my quick'n'dirty version roughly corresponded more or less to the JIT impl.